medusa/restore_node.py

```diff
@@ -74,21 +75,23 @@ def restore_node_locally(config, temp_dir, backup_name, in_place, keep_auth, see
         logging.error('There is nothing to restore')
         sys.exit(0)
 
-    cassandra = Cassandra(config.cassandra)
+    cassandra = Cassandra(config)
 
     # Download the backup
     download_dir = temp_dir / 'medusa-restore-{}'.format(uuid.uuid4())
     logging.info('Downloading data from backup to {}'.format(download_dir))
     download_data(config.storage, node_backup, fqtns_to_restore, destination=download_dir)
 
-    logging.info('Stopping Cassandra')
-    cassandra.shutdown()
-    wait_for_node_to_go_down(config, cassandra.hostname)
+    if not medusa.utils.evaluate_boolean(config.grpc.enabled):
+        logging.info('Stopping Cassandra')
+        cassandra.shutdown()
+        wait_for_node_to_go_down(config, cassandra.hostname)
```

Comment:
This means the gRPC server can exclusively be used for k8s clusters? I thought we would have the option to use it for classic clusters as well and ease automation for folks who prefer invoking Medusa remotely.
Maybe we should have another config setting to indicate whether or not we're dealing with k8s.

Reply:
> I thought we would have the option to use it for classic clusters as well and ease automation for folks who prefer invoking Medusa remotely.

Absolutely. There is nothing Kubernetes specific about the gRPC service. There are integration tests that exercise it, and they do not run in Kubernetes.

> Maybe we should have another config setting to indicate whether or not we're dealing with k8s.

This would be my preference as well. I will add another config setting.

Reply:
I've added a K8s configuration setting now and changed the logic to trigger based on that.